### PR TITLE
[syncd]: Enable port bulk API

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@nvidia.com>

**Summary:**
1. Port bulk API support for `syncd`
2. New test infrastructure for `syncd`
3. New test infrastructure for `VendorSai`
4. `GCOVR` parser hotfix: rollback to stable v5.0

### CODE REVIEW

**Comment by nazariig** [comment]
PR fails due to GCOVR errors:
```
+ gcovr --version
gcovr 5.2

Copyright (c) 2013-2022 the gcovr authors
Copyright (c) 2013 Sandia Corporation.
Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
the U.S. Government retains certain rights in this software.
+ find SAI/meta -name '*.gc*'
+ xargs rm -vf
removed 'SAI/meta/libsaimetadata_la-saimetadata.gcno'
removed 'SAI/meta/libsaimetadata_la-saimetadatautils.gcno'
removed 'SAI/meta/.libs/libsaimetadata_la-saimetadata.gcno'
removed 'SAI/meta/.libs/libsaimetadata_la-saimetadatautils.gcno'
removed 'SAI/meta/.libs/libsaimetadata_la-saimetadatautils.gcda'
removed 'SAI/meta/.libs/libsaimetadata_la-saimetadata.gcda'
removed 'SAI/meta/.libs/libsaimetadata_la-saiserialize.gcno'
removed 'SAI/meta/.libs/libsaimetadata_la-saiserialize.gcda'
removed 'SAI/meta/libsaimetadata_la-saiserialize.gcno'
+ gcovr -r ./ -e '.*/SAI/.*' -e .+/json.hpp -e swss/.+ -e '.*/.libs/.*' -e '.*/debian/.*' --exclude-unreachable-branches --exclude-throw-branches -x --xml-pretty -o coverage.xml
(WARNING) Unrecognized GCOV output for /__w/1/s/meta/SaiSerialize.cpp
	  function sai_serialize_object_id[abi:cxx11](unsigned long) called 7486270 returned NAN % blocks executed 62%
	This is indicative of a gcov output parse error.
	Please report this to the gcovr developers
	at <[https://github.com/gcovr/gcovr/issues>.](https://github.com/gcovr/gcovr/issues%3E.)
(WARNING) Exception during parsing:
	UnknownLineType: function sai_serialize_object_id[abi:cxx11](unsigned long) called 7486270 returned NAN % blocks executed 62%
(ERROR) Exiting because of parse errors.
	You can run gcovr with --gcov-ignore-parse-errors
	to continue anyway.
(ERROR) Trouble processing '/__w/1/s/meta/.libs/libsaimeta_la-SaiSerialize.gcda' with working directory '/__w/1/s/meta'.
Stdout of gcov was >>File 'SaiSerialize.cpp'
Lines executed:94.24% of 2100
Branches executed:93.37% of 3589
Taken at least once:53.39% of 3589
Calls executed:73.83% of 3076
Creating 'SaiSerialize.cpp##74fbe304ca213391ad9d6de89ec6b791.gcov'
```

**Comment by nazariig** [comment]
@kcudnik / @saiarcot895 would you please have a look?

**Comment by nazariig** [comment]
@kcudnik / @saiarcot895 just a kind reminder

File: syncd/tests.cpp [hunk 1]
```diff
@@ -718,145 +718,6 @@ void test_invoke_dump()
     assert(mockCallArg == SAI_FAILURE_DUMP_SCRIPT);
 }
```
Comment (saiarcot895) [inline syncd/tests.cpp:719]: Should `test_invoke_dump` be also moved to `syncd/tests/TestSyncdBrcm.cpp`, or remain in this file?

Comment (nazariig) [reply syncd/tests.cpp:719]: @saiarcot895 ideally - yes

**Review by saiarcot895 — APPROVED** [review]

**Comment by nazariig** [comment]
@saiarcot895 can you please help to merge this PR?

File: unittest/syncd/TestVendorSai.cpp [hunk 2]
```diff
@@ -37,6 +43,96 @@ static sai_service_method_table_t test_services = {
...
+class VendorSaiTest : public ::testing::Test
+{
+public:
+    VendorSaiTest() = default;
+    virtual ~VendorSaiTest() = default;
+
+public:
+    virtual void SetUp() override
+    {
+        m_vsai = std::make_shared<VendorSai>();
```
Comment (kcudnik) [inline unittest/syncd/TestVendorSai.cpp:55]: this is not required in setup, this init and teardown should be in the test itself

Comment (nazariig) [reply unittest/syncd/TestVendorSai.cpp:55]: @kcudnik having the same instruction in each test will lead to code duplication. The whole idea of using this fixture is to minimize amount of the identical code and let the user to focus on API testing. Also, here we want to force a cleanup when the test fails unexpectedly. Please elaborate on why it is a wrong approach?

**Pull request merged**